store/filestore.go

```diff
@@ -134,3 +139,86 @@ func (s *FileStore) LoadCache(walkFn func(item interface{}) error) error {
 		return nil
 	})
 }
+
+func (s *FileStore) Watch(done chan bool) error {
+	if s.watcher != nil {
+		return errors.New("is already watching")
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	defer watcher.Close()
+	if err != nil {
+		return err
+	}
+
+	s.watcher = watcher
+
+	var watcherErr = make(chan error, 1)
+
+	go func() {
+		for {
+			select {
+			case <-done:
+				watcherErr <- nil
+				return
+			case event, ok := <-watcher.Events:
+				if !ok {
+					goto UnableContinueWatching
```

Comment:
Do not use goto. Instead, use break.

https://golang.org/ref/spec#Break_statements